src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -646,6 +649,14 @@ public boolean getUseRegionalS3EndpointsForPresignedURL() {
     return useRegionalS3EndpointsForPresignedURL;
   }
 
+  public String getArrayBindStage() {
+    return arrayBindStage;
+  }
+
+  public void setArrayBindStage(String arrayBindStage) {
+    this.arrayBindStage = String.format("%s.%s.%s", getDatabase(), getSchema(), arrayBindStage);
```

Comment:
Technically a JDBC user could do 
`jdbcSession.getBaseSession.setArrayBindStage()` 

But that is ok because they are just changing the name of the temp stage used for array binds upload  ?

Reply:
Setting the stage name here has no effect on the stage name used in uploads, since that's determined by `session.getSfConnectionHandler().getBindStageName()`. 

It does sort of mess with the logic in `BindUploader` that uses it to check if a different thread has already created and set the bind stage, though. This "flaw" was already pre-existent in the design prior to this refactor in `SFSession`. In any case, this is an undocumented API, and I would consider any user invoking this to be misusing the driver and thus all bets are off.

Reply:
About `Setting the stage name here has no effect on the stage name used in uploads, since that's determined by session.getSfConnectionHandler().getBindStageName()`, can we comment it somewhere to avoid a future incorrect usage?